Models/Terminal.php

```diff
@@ -58,10 +60,10 @@ public static function getTerminals()
      *
      * @throws PrestaShopDatabaseException
      */
-    public static function getActiveTerminals()
+    public static function getActiveTerminals($shop_id = 1)
     {
         return Db::getInstance(_PS_USE_SQL_SLAVE_)->ExecuteS('
-			SELECT * FROM `' . _DB_PREFIX_ . 'altapay_terminals` WHERE active = 1 ORDER BY `display_name` ASC
+			SELECT * FROM `' . _DB_PREFIX_ . 'altapay_terminals` WHERE active = 1 AND shop_id = ' . (int) $shop_id . ' ORDER BY `display_name` ASC
```

Comment:
If the shop_id has characters other than numbers, then casting it to `int` will return `0`, which in itself can be a real `shop_id`, returning results that are not desired.

Please perform proper validation and throw validation error in case the shop_id is not correct.

Reply:
I have updated regarding this in the above comment.